### ISSUE TITLE
stringify: make it async if possible

### ISSUE THREAD

**Issue by aivanov-noveo** [eemeli/yaml#519]
**Is your feature request related to a problem? Please describe.**
If we have a large object (the resulting YAML string is about 3.5MB) then stringifying it could take too much time (and the main thread is blocked on that time).

**Describe the solution you'd like**
Please add `stringifyAsync` method at least to unblock the main thread and at most to use worker threads from Node.js

**Describe alternatives you've considered**
None

**Additional context**
None

**Comment by eemeli** [comment]
Is there a reason why this would need to be functionality provided by the library? Could you not wrap the call yourself in an async wrapper to achieve the same result?

**Comment by aivanov-noveo** [comment]
> Is there a reason why this would need to be functionality provided by the library? Could you not wrap the call yourself in an async wrapper to achieve the same result?

I could but my first thought was like _"If a library has a potentially long sync operation there should be an async-counterpart for that operation (just like for `fs` methods")_
If you think it's a very rare case then you can simply keep this issue with low priority as a potential enhancement or just close it.

**Comment by eemeli** [comment]
It's not so much about the rarity, but about keeping the public API limited. In this case, I think what you're asking for is making something like this a part of it:
```js
import { stringify } from 'yaml';
export const stringifyAsync = async (...args) => stringify(...args);
```

That seems like a rather easy thing for a user to do, should they find it needful. Or do something more complex like use a worker thread, if that's available in their environment.

I'll close this, as I've no current plans of including the above.
